modules/standard/Path.chpl

```diff
@@ -288,17 +288,79 @@ proc file.realPath(): string throws {
   }
 
   /* Determines and returns the longest common path prefix of
-   all the string pathnames provided. If no arguments are specified, returns
-   the empty string.
+   all the string pathnames provided.
+
+   :arg paths: Any number of paths
+   :type paths: `string`
+
+   :return: The longest common path prefix
+   :rtype: `string`
+   */
+
+  proc commonPath(paths: string ...?n): string {
+
+  var result: string = "";    // result string
+  var inputLength = n;   // size of input array
+  var start: int = 1;
+  var end: int = n;
+  var firstPath = paths(1);
+  var flag: int = 0;
+
+  // if input is empty, return empty string.
+  // if input is just one string, return that string as longest common prefix path.
+
+  if inputLength == 0 then {
+    return result;
+  } else if inputLength == 1 then{
+    return firstPath;
+  }
+
+  var prefixArray = firstPath.split(pathSep, -1, false);    // array of resultant prefix string
+
+  var pos = prefixArray.size;   // rightmost index of common prefix
+  var minPathLength = prefixArray.size;
+
+  for i in (start+1)..end do {
```

Comment:
Since `end == n`, and in the varargs version `start+1` will always be `2`, it would be safe to use those values directly for this loop instead of defining `start` and `end` (with a comment to ensure it is clear)